### PR TITLE
RavenDB-15919-2 - Concurrent Subscriptions Performance and test fixes

### PR DESCRIPTION
**Pull request by lastav5**

### CODE REVIEW

**Review by karmeli87 — APPROVED** [review]

File: bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                EnableRevisions();
+
+                var workersList = new List<SubscriptionWorker<dynamic>>();
+                var taskList = new List<Task>();
+                Stopwatch sp = null;
+                for (int i = 0; i < workers; i++)
+                {
+                    var worker = _store.Subscriptions.GetSubscriptionWorker(new SubscriptionWorkerOptions(_subscriptionName)
+                    {
+                        Strategy = workers == 1 ? SubscriptionOpeningStrategy.OpenIfFree : SubscriptionOpeningStrategy.Concurrent,
```
Comment (ayende) [inline bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs]: Let's also test this with a single concurrent worker.

File: bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                    await worker.DisposeAsync().ConfigureAwait(false);
+                }
+                
+                await Task.WhenAll(taskList);
+                
+                return new RunResult
+                {
+                    DocsProccessed = sentCount,
+                    DocsRequested = _batchSize,
+                    ElapsedMs = sp.ElapsedMilliseconds
```
Comment (ayende) [inline bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs]: This is wrong. You are also including the time to *shut down*.
Instead, use `TrySetResult(sp.Elapsed)`

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 3]
```diff
@@ -63,7 +63,7 @@ public class SubscriptionWorker<T> : IAsyncDisposable, IDisposable where T : cla
         /// </summary>
         public event AfterAcknowledgmentAction AfterAcknowledgment;
 
-        internal event Action OnEstablishedSubscriptionConnection;
+        public event Action OnEstablishedSubscriptionConnection;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: should stay internal

Comment (lastav5) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: The benchmark test uses this function, that's why I had to make it public

Comment (ppekrol) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: You need to add that project here
https://github.com/ravendb/ravendb/blob/v5.2/src/CommonAssemblyInfo.Linux.cs
and here
https://github.com/ravendb/ravendb/blob/v5.2/src/CommonAssemblyInfo.Windows.cs

File: bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs (hunk 1 above)
Comment (lastav5) [reply bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs]: Done

File: bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs (hunk 2 above)
Comment (lastav5) [reply bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs]: Done

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs (hunk 3 above)
Comment (gregolsky) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: Actually I'd say that would be valuable to have it public. Seeing just the errors the user cannot be sure it was able to connect in the end or not if no new documents are coming. I think we should make it public.

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**